skimage/filters/ridges.py

```diff
@@ -105,9 +107,139 @@ def _check_sigmas(sigmas):
+        Standard deviation used for the Gaussian kernel, which sets the
+        amount of smoothing in terms of pixel-distances. It is
+        advised to not choose a sigma much than 1.0, otherwise
+        aliasing artifacts may occur.
+    mode : {'constant', 'reflect', 'wrap', 'nearest', 'mirror'}, optional
+        How to handle values outside the image borders.
+    cval : float, optional
+        Used in conjunction with mode 'constant', the value outside
+        the image boundaries.
+    order : {'rc', 'xy'}, optional
+        This parameter allows for the use of reverse or forward order of
+        the image axes in gradient computation. 'rc' indicates the use of
+        the first axis initially (Hrr, Hrc, Hcc), whilst 'xy' indicates the
+        usage of the last axis initially (Hxx, Hxy, Hyy)
+    Returns
+    -------
+    H_elems : list of ndarray
+        Upper-diagonal elements of the hessian matrix for each pixel in the
+        input image. In 2D, this will be a three element list containing [Hrr,
+        Hrc, Hcc]. In nD, the list will contain ``(n**2 + n) / 2`` arrays.
+    --------
+    >>> from skimage.filters.ridges import hessian_matrix_with_Gaussian
+    >>> square = np.zeros((30, 30))
+    >>> square[15, 15] = 1
+    >>> Hrr, Hrc, Hcc = hessian_matrix_with_Gaussian(square, sigma=1)
+    >>> import matplotlib.pyplot as plt
+    >>> plt.imshow(Hcc)
+    >>> plt.colorbar()
+    >>> plt.show()
+    """
+    image = img_as_float(image)
+    float_dtype = _supported_float_type(image.dtype)
+    image = image.astype(float_dtype, copy=False)
+
+    H_elems = []
+    idx = np.arange(image.ndim)
+    # The derivative of an image I convolved with a Gaussian G is
+    #       (d/dx_i)[I*G]
+    # where * indicates a convolution. The distributive property
+    # of derivatives and convolutions allows us to restate this as
+    #        I * dG/dx_i
+    # that is, the convolution of I with the derivative of a Gaussian.
+    # We need to call scipy.ndimage.gaussian_filter with the argument
+    # "order" which indicates the derivative order in the respective
+    # directions, where 0 = just Gaussian smoothing
+    #                   1 = convolve with first derivative of Gaussian
+    #                   etc.
+    # so supplying order=[2, 0] computes the 2nd Gaussian derivative in
+    # the first direction, and just smoothes the field in the second
+    # direction. That corresponds to the lower-right element of the
+    # Hessian matrix, Hcc. This is why below we will call the array
+    # deriv_step{1/2}[::-1] thus in reverse order, because the image
+    # will be in coordinates [(z,)y,x], but we need the Hessian in
+    # order [d^2/dx^2, d^2/(dx*dy), ...], so in reverse order.
+    for deriv_dirs in itertools.combinations_with_replacement(idx, 2):
+        # E.g., for idx=[0, 1] we get deriv_dirs=[0, 0]; [0, 1]; [1, 1]
+
+        deriv_step1 = 1 * (idx == deriv_dirs[0])
+        deriv_step2 = 1 * (idx == deriv_dirs[1])
```

Comment:
```suggestion
        deriv_step1 = (idx == deriv_dirs[0]).astype(int)
        deriv_step2 = (idx == deriv_dirs[1]).astype(int)
```
It is more readable (the intention is explicit) and faster
```python
In [2]: %timeit 1 * np.array([True, False])
1.33 µs ± 3.16 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: %timeit np.array([True, False]).astype(int)
763 ns ± 2.27 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```
:wink:

Reply:
Perfect, thanks for the suggestion!